lib/flight_scheduler/schedulers/fifo_scheduler.rb

```diff
@@ -69,6 +69,13 @@ def run_allocation_loop(candidates)
       .select { |job| job.reason_pending == 'WaitingForScheduling' }
       .each { |job| job.reason_pending = 'Priority' }
 
-    new_allocations
+    excess = partition.nodes.any? { |n| n.state == 'IDLE'}
```

Comment:
This is not quite right either.  Consider the case where there is one node connected to the controller.  The queue is empty.  A job requiring 2 nodes is added.  An insufficient message is sent, immediately followed by an excess message.

This check should be changed to the same as the backfilling check: "if there are zero jobs pending on resources and there are idle nodes we have an excess otherwise we don't"

Reply:
https://github.com/openflighthpc/flight-scheduler-controller/commit/7e2b132082203df553a1e162e4f0d2e864539156